python/PresentationBuilder/slides/RemarkSlide.py

```diff
@@ -80,62 +54,52 @@ def extractName(raw):
   #   ???
   #   This is a comment
   #
-  def __init__(self, name, params, **kwargs):
-    MooseObject.__init__(self, name, params)
+  def __init__(self, params, **kwargs):
+    MooseObject.__init__(self, params)
 
     # Set the Image type
-    self._image_type = kwargs.pop('image_type', 'MarkdownImage')
-
-    # Set the parent object
-    self.parent = self.getParam('_parent') # the SlideSet parent object
-
-    # Get a reference to the factory
-    self._warehouse = self.parent.getParam('_warehouse')
-    self._factory = self.parent.getParam('_factory')
-    self._parser = self.parent.getParam('_parser')
-    self._root = self.parent.getParam('_root')
-
-    # Initialize member variables
-    self.markdown = None                   # storage for parsed markdown (see SlideSet::createSlide)
-    self.index = None                      # slide index, this is only populated and used for table of contents generation
-    self.comments = []                     # comment storage for the current slide
-    self._show_in_contents = self.getParam('show_in_contents')
-    self._title = None
-    self._previous = None
-    self._raw_markdown = self.getParam('markdown')
-    self._format = self.getParam('format')
-
-    # Set the location of PresentationBuilder directory
-    self._source_dir = os.path.abspath(os.path.join(os.path.split(inspect.getfile(self.__class__))[0], '..'))
+    self.__image_type = kwargs.pop('image_type', 'MarkdownImage')
+
+    # Set the parent object, which is the RemarkSlideSet parent object
+    self.parent = self.getParam('_parent')
+
+    # The slide number, this is set by the SlideSetWarehouse::__contents method
+    self.number = 0
+
+    # Get a reference to the factory, parser, and the root node of the input file
+    self.__factory = self.parent.getParam('_factory')
+    self.__parser = self.parent.getParam('_parser')
+    self.__root = self.parent.getParam('_root')
```

Comment:
Yes, double underscore actually makes it private in python, i.e., you can't call it outside of the class. It might be callable in a child class, so it might make it protected. I can't remember.  I agree we need to update our python style. I have been messing around with this stuff for a bit, I am even working on decorators for performing some checks for us to help maintain a consist design and style.